docs/src/doc/dev/adr/0005-use-typescript.stories.mdx

```diff
@@ -0,0 +1,36 @@
+import { Meta } from '@storybook/addon-docs/blocks';
+
+<Meta title="Documentation|dev/ADRs/5) Use typescript" />
+
+# 5) Use TypeScript
+
+Date: 2020-07-16
+
+## Status
+
+Accepted
+
+## Context
+The components library that implements the design system will be implemented in Vue (see ADR #1). In addition to that we disucssied what benefits we can gain by using [TypeScript](https://www.typescriptlang.org/).
+
+There are several products (as of date) in the WMDE realm that were build with Vue, TypeScript and [Decorators](https://github.com/vuejs/vue-class-component): Termbox, Tainted References, Data Bridge, and we anticipate there will be more. This means the components library needs to support TypeScript apps.
+
+Using TypeScript together with Decorators, when developing Vue projects, used to be the recomended way, because Decorators provide full compatibility between Vue and TypeScript. The TypeScript support in Vue has significantly improved since then.
+
+Benefits of using TypeScript:
+ - It can help for internal reuse of behavior (e.g. a toggleable mixin)
+ - It can help for testing
+ - Automatically enables the library to be consumed in TypeScript apps
+
+Downsides of using TypeScript:
+ - Slightly raises the entry barrier for newcommers to Vue
+ - If used with Decorators ([Component](https://github.com/vuejs/vue-class-component), [Prop](https://github.com/kaorun343/vue-property-decorator)) it changes the syntax significantly and the [Vue docs](https://vuejs.org/v2/guide/) become less usable. The Decorators are also not well documented (compared to the [Vue docs](https://vuejs.org/v2/guide/)). The TypeScript support in Vue has improved significantly with time, therefor we might not need the additional help of Decorators to make Vue and TypeScript work well together.
```

Comment:
I think this explains the downsides of Class Components and its decorators really well!

Typo:
```suggestion
 - If used with Decorators ([Component](https://github.com/vuejs/vue-class-component), [Prop](https://github.com/kaorun343/vue-property-decorator)) it changes the syntax significantly and the [Vue docs](https://vuejs.org/v2/guide/) become less usable. The Decorators are also not well documented (compared to the [Vue docs](https://vuejs.org/v2/guide/)). The TypeScript support in Vue has improved significantly with time, therefore we might not need the additional help of Decorators to make Vue and TypeScript work well together.
```